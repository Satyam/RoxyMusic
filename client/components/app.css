.app {
  width: 100%;
  height: 100vh;
}

.wide {
  composes: global(row);
  width: 100%;
  height: 100%;
}

.wide-left-panel {
  composes: global(col-md-6);
  height: 100%;
}

.wide-right-panel {
  composes: global(col-md-6);
  height: 100%;
}

.narrow {
  padding: 1em 0;
}

.menu {
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
}
