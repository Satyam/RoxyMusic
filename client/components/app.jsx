import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import compose from 'recompose/compose';
import Navbar from 'react-bootstrap/lib/Navbar';
import Icon from '_components/misc/icon';
import FoldingToolbar from '_components/misc/foldingToolbar';

import Errors from '_components/misc/errors';
import Loading from '_components/misc/loading';
import Audio from '_components/audio';
import SelectPlayList from '_components/playLists/selectPlayList';
import NowPlaying from '_components/nowPlaying';


// import { remote } from 'electron';
// const db = remote.getGlobal('db');
// console.log(db.all('select * from Albums'));

import styles from './app.css';

const AppComponent = ({ wide, children }) => (
  <div className={styles.app}>
    <Navbar>
      <Navbar.Header>
        <Navbar.Brand>
          <Icon type="heart,music" label="Roxy Music" href="/" title="home" />
        </Navbar.Brand>
      </Navbar.Header>
      <FoldingToolbar>
        {
          BUNDLE === 'cordova'
          ? (
            <Icon button type="retweet" href="/sync" title="Synchonize" />
          )
          : null
        }
        <Icon button type="cog" href="/config" title="Configure" />
      </FoldingToolbar>
    </Navbar>
    <Loading />
    <Errors />
    <SelectPlayList />
    {
      (
        wide
        ? (
          <div className={styles.wide}>
            <div className={styles.wideLeftPanel}>
              { children || (
                <div className={styles.menu}>
                  <Icon
                    button="primary"
                    block
                    type="cd"
                    className={styles.menuItem}
                    href="/albums"
                  >Albums</Icon>
                  <Icon
                    button="primary"
                    block
                    type="user"
                    className={styles.menuItem}
                    href="/artists"
                  >Artists</Icon>
                  <Icon
                    button="primary"
                    block
                    type="music"
                    className={styles.menuItem}
                    href="/songs"
                  >Songs</Icon>
                  <Icon
                    button="primary"
                    block
                    type="list"
                    className={styles.menuItem}
                    href="/playLists"
                  >Play Lists</Icon>
                </div>
              )}
            </div>
            <div className={styles.wideRightPanel}>
              <Audio />
              <NowPlaying />
            </div>
          </div>
        )
        : (
          <div className={styles.narrow}>
            { children || (
              <ul className={styles.menu}>
                <Icon
                  button="primary"
                  block
                  type="cd"
                  className={styles.menuItem}
                  href="/albums"
                >Albums</Icon>
                <Icon
                  button="primary"
                  block
                  type="user"
                  className={styles.menuItem}
                  href="/artists"
                >Artists</Icon>
                <Icon
                  button="primary"
                  block
                  type="music"
                  className={styles.menuItem}
                  href="/songs"
                >Songs</Icon>
                <Icon
                  button="primary"
                  block
                  type="list"
                  className={styles.menuItem}
                  href="/playLists"
                >Play Lists</Icon>
                <Icon
                  button="primary"
                  block
                  type="play"
                  className={styles.menuItem}
                  href="/now"
                >Now Playing</Icon>
              </ul>
            )}
            <Audio />
          </div>
        )
      )
    }
  </div>
);

AppComponent.propTypes = {
  wide: PropTypes.bool,
  children: PropTypes.node,
};

export const mapStateToProps = state => state.config;

export default compose(
  connect(
    mapStateToProps
  )
)(AppComponent);
