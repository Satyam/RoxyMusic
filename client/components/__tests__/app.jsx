import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { AppComponent, mapStateToProps } from '../app';

jest.mock('_store/selectors.js', () => ({
  configSelectors: {
    get: jest.fn((state, name) => state.config[name]),
  },
}));

describe('AppComponent', () => {
  describe('snapshot testing', () => {
    it('with no properties', () => {
      const wrapper = shallow(<AppComponent />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('with wide=true', () => {
      const wrapper = shallow(<AppComponent wide />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('with children', () => {
      const wrapper = shallow(
        <AppComponent>
          <h1>whatever</h1>
        </AppComponent>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('with children and wide=true', () => {
      const wrapper = shallow(
        <AppComponent wide>
          <h1>whatever</h1>
        </AppComponent>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
  describe('mapStateToProps', () => {
    it('should return value from config object cache', () => {
      const initialState = {
        config: {
          wide: true,
        },
      };
      expect(mapStateToProps(initialState, 'wide')).toEqual({ wide: true });
    });
  });
});
