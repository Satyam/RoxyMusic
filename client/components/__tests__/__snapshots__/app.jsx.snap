// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent snapshot testing with children 1`] = `
<div>
  <Uncontrolled(Navbar)
    bsStyle="default"
  >
    <NavbarHeader>
      <NavbarBrand>
        <Icon
          href="/"
          label="Roxy Music"
          title="home"
          type="heart,music"
        />
      </NavbarBrand>
    </NavbarHeader>
    <FoldingToolbar>
      <Icon
        button={true}
        href="/config"
        title="Configure"
        type="cog"
      />
    </FoldingToolbar>
  </Uncontrolled(Navbar)>
  <Connect(LoadingComponent) />
  <Connect(ErrorsComponent) />
  <initStore(Connect(SelectPlayListComponent)) />
  <div>
    <h1>
      whatever
    </h1>
    <initStore(Connect(AudioComponent)) />
  </div>
</div>
`;

exports[`AppComponent snapshot testing with children and wide=true 1`] = `
<div>
  <Uncontrolled(Navbar)
    bsStyle="default"
  >
    <NavbarHeader>
      <NavbarBrand>
        <Icon
          href="/"
          label="Roxy Music"
          title="home"
          type="heart,music"
        />
      </NavbarBrand>
    </NavbarHeader>
    <FoldingToolbar>
      <Icon
        button={true}
        href="/config"
        title="Configure"
        type="cog"
      />
    </FoldingToolbar>
  </Uncontrolled(Navbar)>
  <Connect(LoadingComponent) />
  <Connect(ErrorsComponent) />
  <initStore(Connect(SelectPlayListComponent)) />
  <div>
    <div>
      <h1>
        whatever
      </h1>
    </div>
    <div>
      <initStore(Connect(AudioComponent)) />
      <initStore(Connect(NowPlayingComponent)) />
    </div>
  </div>
</div>
`;

exports[`AppComponent snapshot testing with no properties 1`] = `
<div>
  <Uncontrolled(Navbar)
    bsStyle="default"
  >
    <NavbarHeader>
      <NavbarBrand>
        <Icon
          href="/"
          label="Roxy Music"
          title="home"
          type="heart,music"
        />
      </NavbarBrand>
    </NavbarHeader>
    <FoldingToolbar>
      <Icon
        button={true}
        href="/config"
        title="Configure"
        type="cog"
      />
    </FoldingToolbar>
  </Uncontrolled(Navbar)>
  <Connect(LoadingComponent) />
  <Connect(ErrorsComponent) />
  <initStore(Connect(SelectPlayListComponent)) />
  <div>
    <ul>
      <Icon
        block={true}
        button="primary"
        href="/albums"
        type="cd"
      >
        Albums
      </Icon>
      <Icon
        block={true}
        button="primary"
        href="/artists"
        type="user"
      >
        Artists
      </Icon>
      <Icon
        block={true}
        button="primary"
        href="/songs"
        type="music"
      >
        Songs
      </Icon>
      <Icon
        block={true}
        button="primary"
        href="/playLists"
        type="list"
      >
        Play Lists
      </Icon>
      <Icon
        block={true}
        button="primary"
        href="/now"
        type="play"
      >
        Now Playing
      </Icon>
    </ul>
    <initStore(Connect(AudioComponent)) />
  </div>
</div>
`;

exports[`AppComponent snapshot testing with wide=true 1`] = `
<div>
  <Uncontrolled(Navbar)
    bsStyle="default"
  >
    <NavbarHeader>
      <NavbarBrand>
        <Icon
          href="/"
          label="Roxy Music"
          title="home"
          type="heart,music"
        />
      </NavbarBrand>
    </NavbarHeader>
    <FoldingToolbar>
      <Icon
        button={true}
        href="/config"
        title="Configure"
        type="cog"
      />
    </FoldingToolbar>
  </Uncontrolled(Navbar)>
  <Connect(LoadingComponent) />
  <Connect(ErrorsComponent) />
  <initStore(Connect(SelectPlayListComponent)) />
  <div>
    <div>
      <div>
        <Icon
          block={true}
          button="primary"
          href="/albums"
          type="cd"
        >
          Albums
        </Icon>
        <Icon
          block={true}
          button="primary"
          href="/artists"
          type="user"
        >
          Artists
        </Icon>
        <Icon
          block={true}
          button="primary"
          href="/songs"
          type="music"
        >
          Songs
        </Icon>
        <Icon
          block={true}
          button="primary"
          href="/playLists"
          type="list"
        >
          Play Lists
        </Icon>
      </div>
    </div>
    <div>
      <initStore(Connect(AudioComponent)) />
      <initStore(Connect(NowPlayingComponent)) />
    </div>
  </div>
</div>
`;
